cmd/store/import.go

```diff
@@ -35,79 +36,157 @@ import (
+	fgaClient client.SdkClient,
+	storeData *storetest.StoreData,
+	format authorizationmodel.ModelFormat,
+	storeID string,
+) (*CreateStoreAndModelResponse, error) {
+	store, err := fgaClient.GetStore(context.Background()).Execute()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get store: %w", err)
+	}
 
-		response.Store = client.ClientCreateStoreResponse{
+	response := &CreateStoreAndModelResponse{
+		Store: client.ClientCreateStoreResponse{
 			CreatedAt: store.GetCreatedAt(),
 			Id:        store.GetId(),
 			Name:      store.GetName(),
 			UpdatedAt: store.GetUpdatedAt(),
-		}
+		},
+	}
+
+	authModel := authorizationmodel.AuthzModel{}
+	clientConfig.StoreID = storeID
+
+	err = authModel.ReadModelFromString(storeData.Model, format)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read model: %w", err)
+	}
 
-		authModel := authorizationmodel.AuthzModel{}
-		clientConfig.StoreID = storeID
+	modelWriteRes, err := model.Write(fgaClient, authModel)
+	if err != nil {
+		return nil, fmt.Errorf("failed to write model: %w", err)
+	}
+
+	response.Model = modelWriteRes
+
+	return response, nil
+}
+
+func importStore(
+	clientConfig fga.ClientConfig,
+	fgaClient client.SdkClient,
+	storeData *storetest.StoreData,
+	format authorizationmodel.ModelFormat,
+	storeID string,
+	maxTuplesPerWrite int,
+	maxParallelRequests int,
+	fileName string,
+) (*CreateStoreAndModelResponse, error) {
+	var (
+		response *CreateStoreAndModelResponse
+		err      error
+	)
 
-		err = authModel.ReadModelFromString(storeData.Model, format)
+	if storeID == "" {
+		response, err = createStore(clientConfig, storeData, format, fileName)
 		if err != nil {
-			return nil, err //nolint:wrapcheck
+			return nil, fmt.Errorf("failed to create store: %w", err)
 		}
-
-		modelWriteRes, err := model.Write(fgaClient, authModel)
+	} else {
+		response, err = updateStore(clientConfig, fgaClient, storeData, format, storeID)
 		if err != nil {
-			return nil, fmt.Errorf("failed to write model due to %w", err)
+			return nil, fmt.Errorf("failed to update store: %w", err)
 		}
-
-		response.Model = modelWriteRes
 	}
 
-	fgaClient, err := clientConfig.GetFgaClient()
+	fgaClient, err = clientConfig.GetFgaClient()
 	if err != nil {
-		return nil, fmt.Errorf("failed to initialize FGA Client due to %w", err)
+		return nil, fmt.Errorf("failed to initialize FGA Client: %w", err)
 	}
 
 	writeRequest := client.ClientWriteRequest{
 		Writes: storeData.Tuples,
 	}
 
-	_, err = tuple.ImportTuples(fgaClient, writeRequest, maxTuplesPerWrite, maxParallelRequests)
-	if err != nil {
-		return nil, err //nolint:wrapcheck
+	// Initialize progress bar
+	bar := progressbar.NewOptions(len(storeData.Tuples),
+		progressbar.OptionSetDescription("Importing tuples"),
+		progressbar.OptionShowCount(),
+		progressbar.OptionSetWidth(progressBarWidth),
+		progressbar.OptionClearOnFinish(),
+		progressbar.OptionFullWidth(),
+		progressbar.OptionThrottle(progressBarThrottleValue*progressBarSleepDelay),
```

Comment:
I think we might need the sleep between updates that was previously implemented, currently importing 1000 tuples at one tuple per write looks like the below for me, lots of flickering and hard to see the progress/timing

https://github.com/openfga/cli/assets/8705251/faf0f8d9-6c54-45ef-9169-205ca26ff808


However, if I add a 5 ms delay between iterations it looks a lot smoother, although obviously it does mean a longer import time.

https://github.com/openfga/cli/assets/8705251/7d2340e3-3335-4cfa-a61a-df915932a251

Reply:
Resolved in 659bd39